[Debug][JSC] Fix debug build

csaavedra opened the pull request:
#### 8d9ebc9dbcc637e5ef3f37fccc0502ac5d141909
<pre>
[Debug][JSC] Fix debug build
<a href="https://bugs.webkit.org/show_bug.cgi?id=268904">https://bugs.webkit.org/show_bug.cgi?id=268904</a>

Reviewed by Mark Lam.

lowWidth is declared unsigned, no need for negative assert
check.

* Source/JavaScriptCore/b3/B3LowerToAir.cpp:

Canonical link: <a href="https://commits.webkit.org/274305@main">https://commits.webkit.org/274305@main</a>
</pre>

hyjorc1 commented on Source/JavaScriptCore/b3/B3LowerToAir.cpp:
```diff
@@ -3384,10 +3384,10 @@ class LowerToAir {
+                int64_t highWidth = highWidthValue->asInt();
+                int64_t lowWidth = lowWidthValue->asInt();
+                int64_t datasize = opcode == ExtractRegister32 ? 32 : 64;
                 // Note that when `lowWidth == datasize` we cannot turn it to `MOV Rd Rn` since
```
These values cannot be negative. So, we should use unsigned types here. In order to get grid of unexpected warning, we should explicitly cast them to `uint64_t`.

hyjorc1 replied:
> These values cannot be negative.

I mean `highWidth` and `lowWidth` can have negative values. The intent here is to find the pattern that `0 <= lowWidth < datasize` and `0 < highWidth <= datasize` and `lowWidth + highWidth == datasize`.

nvm, we checked these before
```
if (!imm(highWidthValue) || highWidthValue->asInt() < 0)
      return false;
if (!imm(lowWidthValue) || lowWidthValue->asInt() < 0)
      return false;
```

hyjorc1 replied:
Yeah, we should do
```
uint64_t highWidth = static_cast<uint64_t>(highWidthValue->asInt());
uint64_t lowWidth = static_cast<uint64_t>(lowWidthValue->asInt());
```

hwti replied:
The warning is not on these lines.
If you want to keep the `uint64_t` variables, the assert needs a cast :
```
ASSERT(0 <= static_cast<int64_t>(lowWidth) && lowWidth < datasize);
```

hyjorc1 replied:
I see, let's keep the unsigned types and change the assertion to
```
ASSERT(lowWidth < datasize);
```
since `uint64_t lowWidth` must be `>= 0`.

csaavedra replied:
OK, let's go back to the old patch then.

hyjorc1 reviewed (approved):
r=me

MenloDorian reviewed (approved):
r=me

csaavedra committed:
[Debug][JSC] Fix debug build
https://bugs.webkit.org/show_bug.cgi?id=268904

Reviewed by Mark Lam.

lowWidth is declared unsigned, no need for negative assert
check.

* Source/JavaScriptCore/b3/B3LowerToAir.cpp:

Canonical link: https://commits.webkit.org/274305@main
